src/Microsoft.Azure.SignalR.Common/ServiceConnections/ServiceConnectionBase.cs

```diff
@@ -180,6 +189,43 @@ public Task StopAsync()
             return Task.CompletedTask;
         }
 
+        public async Task CloseAsync()
+        {
+            if (_closeIncomingLock.Wait(0))
+            {
+                // incoming task hasn't started yet so we don't need to wait.
+                return;

```

Comment:
release before return, and is it necessary?

Reply:
Yes, the lock should be held by the incoming task and if we got this here, it means the incoming task hasn't started so we could simply return.